### PR TITLE
[on-prem] Bug 2089775: Fix regexp in keepalived script chk_default_ingress.sh

### PR DESCRIPTION
**Pull request by nvsmirnov**
Fixes: #3155

**- What I did**
Changed regexp in template. Really I didn't even try to build MCO, just created machineconfig override to test if it works :-)

**- How to verify it**
Apply fix on cluster and see if destination file contains correct grep command, and keepalived keeps ingress VIP on same node that runs ingress pod.
Ideally one can check it on cluster that has nodes with IP as described in issue #3155.

**- Description for the changelog**
Fixed keepalived check for ingress VIP

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
As [noted](https://github.com/openshift/machine-config-operator/issues/3155#issuecomment-1135107747) on the issue, this will need a bugzilla.

**Comment by nvsmirnov** [comment]
Opened bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2089775

File: templates/common/on-prem/files/keepalived-script-default-ingress.yaml
Comment (cybertron) [inline templates/common/on-prem/files/keepalived-script-default-ingress.yaml:6]: Can we combine the two greps here so we have correct boundaries at both the start and end of the IP? I accidentally pushed a duplicate fix in https://github.com/openshift/machine-config-operator/pull/3164 but since you proposed this first I'd prefer to merge it under your PR.

Comment (nvsmirnov) [reply templates/common/on-prem/files/keepalived-script-default-ingress.yaml:6]: Yes, of course. Strictly saying, the form with grep still not exact (because dot is special character)
It is better process with jq, I'll figure out soon how it is better to do this with jq filters :-)

Comment (nvsmirnov) [reply templates/common/on-prem/files/keepalived-script-default-ingress.yaml]: @cybertron, I've changed PR so it now uses jq instead of grep, and this should work more precise.

Comment (nvsmirnov) [reply templates/common/on-prem/files/keepalived-script-default-ingress.yaml]: ...the only problem is I can't check here if this template will render correctly to destination file.

What I did is checked on real OKD node that this works:

`oc --kubeconfig /var/lib/kubelet/kubeconfig get ep -n openshift-ingress router-internal-default -o json | jq -e '.subsets[].addresses | map( select (.ip == "IP_ADDRESS") )[0]'; echo $?`

If `IP_ADDRESS` is in oc output, it sets exit status 0, and non-zero if it is not in output.

**Comment by cybertron** [comment]
/ok-to-test

Thanks, I will also pull this down locally so I can poke around the environment after deployment and make sure everything looks to be doing what it's supposed to.

**Review by cybertron — CHANGES_REQUESTED** [review]
Unfortunately it looks like jq is not available in the keepalived image, which may explain why we didn't use it originally.

Grep should be safe to use in this case if we choose the regex correctly. If a . matched a number incorrectly, then the next . in the IP would not match. Since we know the value here is an IP, we don't have to worry about matching something like 1a1b1c1 to 1.1.1.1 since that would be invalid for the field in the first place.

**Comment by nvsmirnov** [comment]
@cybertron, oh yes. I changed back to one grep, thank you.

**Review by mandre — COMMENTED** [review]
The e2e-openstack failure is unrelated and can safely be ignored.
/lgtm
/cherry-pick release-4.10

**Comment by cybertron** [comment]
/lgtm

Thanks!

**Review by sinnykumari — APPROVED** [review]
adding approval, all tests are green.
Please remove hold when ready to merge

**Comment by cybertron** [comment]
/hold cancel

We have lgtm from multiple platforms and all of the tests are passing. We should be good.

**Pull request merged**
